client/src/js/components/courseEvals/CourseEvals.js

```diff
@@ -0,0 +1,300 @@
+           "classroom":"undecided",
+           "location":"-1-undecided"
+        }
+      ],
+      "keywords":[
+        "IPSE"
+      ],
+      "lang":"EN",
+      "keys":[
+        {
+           "school":"FSE",
+           "key":"26GC02300301202026MC03300926"
+        }
+      ]
+    }
+  ];
+}
+
+const getCourseEvals = (evaluations, courseKey) => evaluations.filter(e => e.course_key === courseKey);
+
+class CourseEvals extends React.Component {
+  state = {
+    thisCourse: {},
+    relatedCourses: [],
+    evaluations: [],
+    thisCourseEvals: [],
+    satisfaction: 0,
+    difficulty: 0,
+    benefit: 0,
+    isLoaded: false
+  };
+
+  async componentDidMount () {
+    const courseID = qs.parse(this.props.location.search, { ignoreQueryPrefix: true }).courseID;
+    const coursesList = await getAllCourses();
+    const thisCourse = getCourse(coursesList, courseID);
+    const thisCourseCode = getCourseCode(thisCourse);
+    const thisCourseKey = getCourseKey(thisCourse);
+    const relatedCourses = getRelatedCourses(coursesList, thisCourseCode, thisCourseKey);
+    const evaluations = await getAllEvals(thisCourseCode);
+    const thisCourseEvals = getCourseEvals(evaluations, thisCourseKey);
+    let satisfactionSum = 0, difficultySum = 0, benefitSum = 0;
+    thisCourseEvals.forEach(evaluation => {
+      satisfactionSum += evaluation.satisfaction;
+      difficultySum += evaluation.difficulty;
+      benefitSum += evaluation.benefit;
+    });
+    const satisfaction = Math.round(satisfactionSum / thisCourseEvals.length * 2) / 2;
+    const difficulty = Math.round(difficultySum / thisCourseEvals.length * 2) / 2;
+    const benefit = Math.round(benefitSum / thisCourseEvals.length * 2) / 2;
+
+    this.setState({
+      thisCourse: thisCourse,
+      relatedCourses: relatedCourses,
+      evaluations: evaluations,
+      thisCourseEvals: thisCourseEvals,
+      satisfaction: satisfaction,
+      difficulty: difficulty,
+      benefit: benefit,
+      isLoaded: true
+    });
+  }
+
+  render () {
+    const { thisCourse, relatedCourses, evaluations, thisCourseEvals, satisfaction, difficulty, benefit, isLoaded } = this.state;
+    return (
+      <RowWrapper>
+        <Helmet>
+          <title>WasedaTime - Course Evaluations</title>
+          <meta
+            name="description"
+            content="Syllabus Searching at Waseda University."
+            />
+          <meta property="og:title" content="WasedaTime - Course Evaluations" />
+          <meta
+            property="og:description"
+            content="Syllabus Searching at Waseda University."
+            />
+          <meta property="og:site_name" content="WasedaTime - Course Evaluations" />
+        </Helmet>
+
+        <LongWrapper>
+          <ExtendedOverlay>
+            <div>
+              <br />
+              {
+                isLoaded && <FetchedCourseItem searchTerm={""} searchLang={"jp"} course={thisCourse} isInCourseEvalsPage={true} />
```

Comment:
In the syllabus component,`searchLang` is determined by the keyword that the user inputs in the search bar. 
Here I suggest to get it from the view evaluation button: href={`/courseEvals?courseID=${syllabusId}&searchLang=${searchLang}`} and then parse the url.